yb-voyager/cmd/live_migration.go

```diff
@@ -106,21 +110,87 @@ func (event *Event) getDeleteStmt(targetSchema string) string {
 //==================================================================================
 
 func streamChanges(connPool *tgtdb.ConnectionPool, targetSchema string) error {
-	queueFilePath := filepath.Join(exportDir, "data", "queue.json")
-	log.Infof("Streaming changes from %s", queueFilePath)
-	file, err := os.OpenFile(queueFilePath, os.O_CREATE, 0640)
+	cdcDirPath := filepath.Join(exportDir, "data", "cdc")
+	archiveDirPath := filepath.Join(exportDir, "data", "cdc", "archive")
+	err := os.MkdirAll(archiveDirPath, 0755)
 	if err != nil {
-		return fmt.Errorf("error opening file %s: %v", queueFilePath, err)
+		return fmt.Errorf("error creating archive dir: %v", err)
+	}
+
+	// keep looking for segments continuously
+	// and stream changes from segments one by one in sequence
+	for {
+		dirEntries, err := os.ReadDir(cdcDirPath)
+		if err != nil {
+			return fmt.Errorf("error reading cdc dir: %v", err)
+		}
+
+		var queueSegmentFiles []string
+		for _, dirEntry := range dirEntries {
+			if !dirEntry.IsDir() && strings.HasSuffix(dirEntry.Name(), ".ndjson") {
+				queueSegmentFiles = append(queueSegmentFiles, dirEntry.Name())
+			}
+		}
```

Comment:
Nooo.... not again!!! :-)

Please create proper "classes" with proper "methods". Something like:
```
type EventQueue struct {

}
func NewEventQueue(exportDir string) *EventQueue
func (eq *EventQueue) Init() error
func (eq *EventQueue) GetPendingSegments() ([]*EventQueueSegment, error)

type EventQueueSegment struct {
    Index int
    AbsFilePath string
}
func (seg *EventQueueSegment) Open() (io.ReadCloser, error)
```
Almost always represent every "external" thing as a separate class (e.g. SourceDB, TargetDB, Workspace, EventQueue, etc.)